File: News-Android-App/src/main/java/de/luhmer/owncloudnewsreader/helper/ForegroundListener.java
package de.luhmer.owncloudnewsreader.helper;

import android.app.Activity;
import android.app.Application;
import android.os.Bundle;

import androidx.annotation.NonNull;

public class ForegroundListener implements Application.ActivityLifecycleCallbacks {

    private static int numStarted;

    public static boolean isInForeground() {
        return numStarted > 0;
    }

    @Override
    public void onActivityCreated(@NonNull Activity activity, Bundle savedInstanceState) {
        // do nothing
    }

    @Override
    public void onActivityStarted(@NonNull Activity activity) {
        numStarted++;
    }

    @Override
    public void onActivityResumed(@NonNull Activity activity) {
        // do nothing
    }

    @Override
    public void onActivityPaused(@NonNull Activity activity) {
        // do nothing
    }

    @Override
    public void onActivityStopped(@NonNull Activity activity) {
        numStarted--;
    }

    @Override
    public void onActivitySaveInstanceState(@NonNull Activity activity, @NonNull Bundle outState) {
        // do nothing
    }

    @Override
    public void onActivityDestroyed(@NonNull Activity activity) {
        // do nothing
    }
}

File: News-Android-App/src/main/java/de/luhmer/owncloudnewsreader/helper/ForegroundListener.kt
package de.luhmer.owncloudnewsreader.helper

import android.app.Activity
import android.app.Application.ActivityLifecycleCallbacks
import android.os.Bundle

class ForegroundListener : ActivityLifecycleCallbacks {
    override fun onActivityCreated(
        activity: Activity,
        savedInstanceState: Bundle?,
    ) {
        // do nothing
    }

    override fun onActivityStarted(activity: Activity) {
        numStarted++
    }

    override fun onActivityResumed(activity: Activity) {
        // do nothing
    }

    override fun onActivityPaused(activity: Activity) {
        // do nothing
    }

    override fun onActivityStopped(activity: Activity) {
        numStarted--
    }

    override fun onActivitySaveInstanceState(
        activity: Activity,
        outState: Bundle,
    ) {
        // do nothing
    }

    override fun onActivityDestroyed(activity: Activity) {
        // do nothing
    }

    companion object {
        private var numStarted = 0
        val isInForeground: Boolean
            get() = numStarted > 0
    }
}
